pkg/policy/api/rule.go

```diff
@@ -0,0 +1,234 @@
+// Copyright 2016-2017 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package api
+
+import (
+	"github.com/cilium/cilium/pkg/labels"
+)
+
+// Rule is a policy rule which must be applied to all endpoints which match the
+// labels contained in the endpointSelector
+//
+// Multiple rules can select the same endpoint in which case all rules are
+// accumulated additively
+//
+// The rule is split into an ingress section which contains all rules
+// applicable at ingress, and an egress section applicable at egress.
```

Comment:
I would also specify that for a flow between two endpoints to be allowed by Cilium, it must be allowed by both the egress section of the source endpoint and by the ingress section of the destination endpoint.